Jenkinsfile

```diff
@@ -91,14 +91,50 @@ pipeline {
                             equals expected: true, actual: params.FORCE_REDEPLOY_WEB
                         }
                     }
-                    steps {
-                        echo 'Building research-hub-web project'
-                        dir("research-hub-web") {
-                            echo 'Installing research-hub-web dependencies'
-                            sh "npm install"
-
-                            echo 'Building for production'
-                            sh "npm run build -- -c ${BRANCH_NAME}"
+                    stages {
+                        stage ('Caching new node_modules folder') {
+                            when {
+                                anyOf {
+                                    changeset "**/research-hub-web/package.json"
+                                    equals expected: true, actual: params.FORCE_REDEPLOY_WEB
+                                }
+                            }
+                            steps {
+                                echo 'Installing research-hub-web dependencies.'
+                                dir("research-hub-web") {
+                                    sh "npm install"
+                                    sh "mkdir -p ${HOME}/research-hub-web/"
+                                    sh "tar cvfz ./node_modules.tar.gz node_modules" // Cache new node_modules/ folder
+                                    script {
+                                        archiveArtifacts artifacts: "node_modules.tar.gz", onlyIfSuccessful: true
+                                    }
+                                }
+                            }
+                        }
+                        stage ('Unzipping existing cached node_modules.') {
+                            when {
+                                not {
+                                    anyOf {
+                                        changeset "**/research-hub-web/package.json"
+                                    }
+                                }
+                            }
+                            steps {
```

Comment:
These changes had to be done on sandbox in order to test the deployment functionality.